### DISCUSSION TITLE
Can't run from crontab

### DISCUSSION THREAD

**Discussion by AndrewCatford** [teler-sh/teler#157] [Q&A]
I have Teler (v1.2.2) processing logs generated by a Citrix ADC (formerly NetScaler).
Everything works fine interactively, and from within a script, however when I place that script into cron (for both root and unprivileged user), teler responds with:

[WRN] No logs analyzed, did you write log format correctly?

It is not an issue with paths or permissions, and I have the script echoing the exact command line that is going to be used to a temp file and everything is correct: all paths are qualified, all arguments are present, the input log file exists and is readable, the would-be output path is writeable (and in fact Teler creates an empty file with the expected name).

Is there some dependancy on an interractive TTY?
Is it possible to turn on some kind of debugging?

Thank you.

**Comment by dwisiswant0** [comment]
That means the `log_format` you configured on doesn't match. Simple question, are you use the same Teler configuration file (like you run interactively) for cron command, @AndrewCatford?

**Reply by AndrewCatford** [reply]
Yes, it's the exact same configuration file. It works perfectly when run on the command line.

I tried running the command from cron using 'strace -ff -o [...]' and none of the strace logs show any of the Teler processes/threads actually processing anything from the input log file. I'd have expected to see a read() and at least the first line of the log file appear.

I'm not familiar with Go at all, but is it possible the way I am invoking Teler is somehow slipping through the stdin/input file logic in internal/runner/runner.go on lines 112-120?

**Reply by dwisiswant0** [reply]
If may I know, please elaborate on both cron & interactive commands consumed - so that I can reproduce them too, @AndrewCatford.

**Reply by AndrewCatford** [reply]
Apologies for the delay in responding, I was away from work for a few days.

I am invoking Teler with the command:
/usr/local/bin/teler -c /usr/local/etc/teler.yaml -i [input log filename] --json -o [output filename] >/tmp/run.txt 2>&1

Normally /tmp/run would be /dev/null, but I'm using that to capture what teler is printing out.

I have attached 20 lines of log file (with the domain of the vhosts changed) and the teler.yaml file. I had to rename teler.yaml to yaml.txt before github would let me upload it.

Let me know if you require a larger data set.

[sample.log](https://github.com/kitabisa/teler/files/6659451/sample.log)
[yaml.txt](https://github.com/kitabisa/teler/files/6659456/yaml.txt)

**Reply by dwisiswant0** [reply]
Aight. I guess the root cause of "no logs analyzed" as:
- If you use the same command for CLI and cron, please note the Teler configuration file _(specific to_ `log_format`_)_ is using the correct version _(from your CLI)_, or
- The output of log-lines from cron is **NOT** same as CLI, so it'll **PASS** the `log_format` configuration to analyzed.

Since you only provided a sample log from one-side _(either CLI or cron)_, please attach both for further investigation - or you can verify it yourself, @AndrewCatford.

**Reply by AndrewCatford** [reply]
That was the whole point of my original question...
It is the exact same command operating on the exact same log file from the exact same location, and yet somehow when run from cron it doesn't work.
I guess the next step is for me to try and learn Go and add some debugging information to try and make sense of it. It's not a configuration issue.

**Reply by ggaitandv** [reply]
Try:
cat /var/log/nginx/access.log | teler -c /root/teler/teler.yaml
